src/expr/core/src/expr/expr_udf.rs

```diff
@@ -221,3 +246,49 @@ pub(crate) fn get_or_create_client(link: &str) -> Result<Arc<ArrowFlightUdfClien
         Ok(client)
     }
 }
+
+/// Get or create a wasm runtime.
+///
+/// Runtimes returned by this function are cached inside for at least 60 seconds.
+/// Later calls with the same link will reuse the same runtime.
```

Comment:
Are we essentially trying to reuse the same runtime for different actors/tasks for the same job? BTW, what's the benefits or overheads for reusing or not?

Reply:
The runtime supports parallel calls from multiple threads. A WASM instance can only be called from a single thread (for now, [multi-thread support](https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md) is on the way), but we can use an instance pool to share and recycle instances across threads. The benefits of reuse are primarily saving memory, reducing the overhead of duplicate JITs and repeated downloads from object store.